app/components/markdown/markdown_link/markdown_link.js

```diff
@@ -40,19 +44,33 @@ export default class MarkdownLink extends PureComponent {
             return;
         }
 
-        const match = matchPermalink(url, serverURL) || matchPermalink(url, siteURL) || matchPermalink(url, '');
-
-        if (match) {
-            const teamName = match[1];
-            const postId = match[2];
-            onPermalinkPress(postId, teamName);
-        } else {
+        switch (getDeepLinkType(url)) {
```

Comment:
Perhaps we could combine `getDeepLinkType` and `matchChannelLink`/`matchPermalink` somehow? Something like this might simplify the logic since we use it in multiple places
```javascript
function matchDeepLink(url, serverURL, siteURL) {
    const linkRoot = `(?:${escapeRegex(serverURL)}|${escapeRegex(siteURL)})?`;

    let match = new RegExp('^' + linkRoot + '\\/([^\\/]+)\\/channels\\/(\\S+)').exec(link);
    if (match) {
        return {type: DeepLinkTypes.CHANNEL, teamName: match[1], channelName: match[2]};
    }

    match = new RegExp('^' + linkRoot + '\\/([^\\/]+)\\/pl\\/(\\w+)').exec(link);
    if (match) {
        return {type: DeepLinkTypes.PERMALINK, teamName: match[1], postId: match[2]};
    }

    return null;
}
```

0/5 on if we actually change it though